templates/.kokoro/build.sh

```diff
@@ -23,11 +23,41 @@ cd ${scriptDir}/..
 # include common functions
 source ${scriptDir}/common.sh
 
+# Find the files changed from when the PR branched to the last commit
+# Filter for java modules and get all the unique elements
+# grep returns 1 (error code) and exits the pipeline if there is no match
+# If there is no match, it will return true so the rest of the commands can run
+modified_files=$(git diff --name-only $KOKORO_GITHUB_PULL_REQUEST_COMMIT $KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH)
```

Comment:
Use `git diff` to find what changed between the two branches. No longer tries to find the `Base Commit` since the `Target Branch` could have added commits that have been committed in.

Reply:
Memo: Later, we may want to run integration tests as nightly builds.

Reply:
I think this works as long as there are no interdependencies among the modules, which I think is the case with pure GAPICs. However, it might not work in the general case of multi-module Maven projects.